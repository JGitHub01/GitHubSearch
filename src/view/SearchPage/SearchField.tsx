interface ISearchFieldProps {
  keyword?: string
  onKeywordChange?: (keyword: string) => void
}
export default function SearchField(props: ISearchFieldProps) {
  return (
    <input type='text' data-testid='search-field'
      className='github-search__keyword'
      value={props.keyword}
      onChange={(e) => {
        props.onKeywordChange && props.onKeywordChange(e.target.value);
      }}
    />
  );
}
