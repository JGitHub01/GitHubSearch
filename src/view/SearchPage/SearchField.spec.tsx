import {render} from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import SearchField from './SearchField';

describe('SearchField', () => {
  it('Should render the basic snapshot view', () => {
    const {getByTestId} = render(<SearchField />);
    const searchField = getByTestId('search-field');
    expect(searchField).toMatchSnapshot();
  });

  it('Should call onKeywordChange with the entered text when a keyword is entered', () => {
    const callback = jest.fn();
    const {getByTestId} = render(<SearchField onKeywordChange={callback}/>);
    const searchField = getByTestId('search-field');
    userEvent.type(searchField, "tetris");
    expect(callback).toBeCalledWith("tetris");
  })
});
