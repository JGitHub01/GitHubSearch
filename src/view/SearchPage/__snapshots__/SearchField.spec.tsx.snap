// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchField Should render the basic snapshot view 1`] = `
<input
  class="github-search__keyword"
  data-testid="search-field"
  type="text"
  value=""
/>
`;
